### PR TITLE
Add docker 23.0.0 and 23.0.1

### PR DESCRIPTION
**Pull request by doflamingo721**
Added latest docker 23.0.x versions as part of https://github.com/rancher/rancher/issues/40731

### CODE REVIEW

**Review by superseb — CHANGES_REQUESTED** [review]
Changes are not tested in CI:

```
==> Found 23.0.0.sh
==> Skipping 23.0.0 for oraclelinux:7
==> Skipping 23.0.0 for oraclelinux:8
==> Skipping 23.0.0 for centos:centos7
==> Skipping 23.0.0 for centos:centos8
==> Skipping 23.0.0 for ubuntu:18.04
==> Skipping 23.0.0 for ubuntu:20.04
==> Skipping 23.0.0 for ubuntu:22.04
==> Skipping 23.0.0 for rockylinux:8
==> Waiting for all tests for Docker version 23.0.0 are complete
==> Found 23.0.1.sh
==> Skipping 23.0.1 for oraclelinux:7
==> Skipping 23.0.1 for oraclelinux:8
==> Skipping 23.0.1 for centos:centos7
==> Skipping 23.0.1 for centos:centos8
==> Skipping 23.0.1 for ubuntu:18.04
==> Skipping 23.0.1 for ubuntu:20.04
==> Skipping 23.0.1 for ubuntu:22.04
==> Skipping 23.0.1 for rockylinux:8
==> Waiting for all tests for Docker version 23.0.1 are complete
```

**Comment by doflamingo721** [comment]
@superseb I updated the test-docker script to include 23.0 docker versions. Can you please review it again?

**Review by superseb — DISMISSED** [review]

**Review by snasovich — CHANGES_REQUESTED** [review]

File: scripts/test-docker [hunk 1]
```diff
@@ -13,14 +13,14 @@ while true; do
...
-test_docker_versions["oraclelinux:8"]="^20.10"
-test_docker_versions["rockylinux:8"]="^20.10.[8-9]|^20.10.1[0-9]|^20.10.2[0-9]"
+test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10|^23.0.[0-1]"
+test_docker_versions["ubuntu:20.04"]="^^19.03.15$|^20.10|^23.0.[0-1]"
+test_docker_versions["ubuntu:22.04"]="^20.10.1[3-9]|^20.10.2[0-9]|^23.0.[0-1]"
+test_docker_versions["centos:centos7"]="^19.03.15$|^20.10|^23.0.[0-1]"
+test_docker_versions["centos:centos8"]="^20.10.[7-9]|^20.10.1[0-9]|^20.10.2[0-9]|^23.0.[0-1]"
+test_docker_versions["oraclelinux:7"]="^19.03.15$|^20.10|^23.0.[0-1]"
+test_docker_versions["oraclelinux:8"]="^20.10|^23.0.[0-1]"
+test_docker_versions["rockylinux:8"]="^20.10.[8-9]|^20.10.1[0-9]|^20.10.2[0-9]|^23.0.[0-1]"
```
[the remark this answers is not in the document]
Comment (doflamingo721) [reply scripts/test-docker]: Updated the regex to include future 23.0 releases. Can you please review it again?

File: scripts/test-docker [hunk 2]
```diff
@@ -13,14 +13,14 @@ while true; do
...
 
-test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10"
-test_docker_versions["ubuntu:20.04"]="^^19.03.15$|^20.10"
-test_docker_versions["ubuntu:22.04"]="^20.10.1[3-9]|^20.10.2[0-9]"
-test_docker_versions["centos:centos7"]="^19.03.15$|^20.10"
-test_docker_versions["centos:centos8"]="^20.10.[7-9]|^20.10.1[0-9]|^20.10.2[0-9]"
-test_docker_versions["oraclelinux:7"]="^19.03.15$|^20.10"
-test_docker_versions["oraclelinux:8"]="^20.10"
-test_docker_versions["rockylinux:8"]="^20.10.[8-9]|^20.10.1[0-9]|^20.10.2[0-9]"
+test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10|^23.0.([0-9]|10)"
```
Comment (snasovich) [inline scripts/test-docker:16]: Looks like this only includes 0 through 10. Why not simply go with below, same as we do for 20.10:
```suggestion
test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10|^23.0"
```

**Review by snasovich — CHANGES_REQUESTED** [review]

File: scripts/test-docker (hunk 2 above)
Comment (doflamingo721) [reply scripts/test-docker]: @snasovich I updated to include all future 23.0 docker releases. Can you please review it again?

**Review by snasovich — APPROVED** [review]
@doflamingo721 , thank you. LGTM.

**Pull request merged**